### PR TITLE
fix reporting variables in the community report

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
#### Summary
Before 👇 
Screenshot 2024-06-08 033158

After ☝️ 

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.
- [x] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by toninis** [comment]
Fixed in favor of https://github.com/mattermost/mattermost-mobile/pull/7994/commits/e92c1b7369eed6b3124193f9aeb855d8636d4612

File: .github/workflows/e2e-detox-scheduled.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Detox E2E Tests (Scheduled)
 
 on:
   schedule:
-    - cron: "0 0 * * *" # every day at midnight
+    - cron: "0 0 * * 4,5" # Wednesday and Thrusday midnight
```
Comment (saturninoabril) [inline .github/workflows/e2e-detox-scheduled.yml:5]:
```suggestion
    - cron: "0 0 * * 4,5" # Wednesday and Thursday midnight
```

File: .github/workflows/e2e-detox-template.yml [hunk 2]
```diff
@@ -37,6 +33,11 @@ on:
         required: false
         type: boolean
         default: false
+      record_tests_in_zephyr:
+        description: "Record Test results in Zephyr. (Nightly and Release runs)"
```
Comment (saturninoabril) [inline .github/workflows/e2e-detox-template.yml:37]:
```suggestion
        description: "Record test results in Zephyr, typically for nightly and release runs"
```

**Review by saturninoabril — APPROVED** [review]
Thanks @yasserfaraazkhan! Left couple of requests.

File: .github/workflows/e2e-detox-template.yml [hunk 3]
```diff
@@ -48,16 +59,20 @@ on:
...
+  DEVICE_OS_VERSION: ${{ inputs.ios_device_os_name }}
   DETOX_AWS_S3_BUCKET: "mattermost-detox-report"
   HEADLESS: "true"
   TYPE: ${{ inputs.run-type }}
   PULL_REQUEST: "https://github.com/mattermost/mattermost-mobile/pull/${{ github.event.number }}"
-  SITE_1_URL: ${{ inputs.MM_TEST_SERVER_URL || secrets.MM_MOBILE_E2E_SITE_1_URL }}
-  SITE_2_URL: ${{ secrets.MM_MOBILE_E2E_SITE_2_URL }}
-  SITE_3_URL: ${{ secrets.MM_MOBILE_E2E_SITE_3_URL }}
-  ZEPHYR_ENABLE: false
+  SITE_1_URL: "https://mobile-e2e-site-1.test.mattermost.cloud"
```
Comment (toninis) [inline .github/workflows/e2e-detox-template.yml:72]: You removed the CMT functionality . We need to change the SITE_1 URL for the CMT tests 🚀 
```suggestion
  SITE_1_URL: ${{ inputs.MM_TEST_SERVER_URL || "https://mobile-e2e-site-1.test.mattermost.cloud" }} 
```

File: .github/workflows/e2e-detox-template.yml [hunk 4]
```diff
@@ -48,16 +59,20 @@ on:
 env:
   AWS_REGION: "us-east-1"
   ADMIN_EMAIL: ${{ secrets.MM_MOBILE_E2E_ADMIN_EMAIL }}
-  ADMIN_USERNAME: ${{ inputs.MM_TEST_USER_NAME || secrets.MM_MOBILE_E2E_ADMIN_USERNAME }}
-  ADMIN_PASSWORD: ${{ inputs.MM_TEST_PASSWORD || secrets.MM_MOBILE_E2E_ADMIN_PASSWORD }}
+  ADMIN_USERNAME: ${{ secrets.MM_MOBILE_E2E_ADMIN_USERNAME }}
+  ADMIN_PASSWORD: ${{ secrets.MM_MOBILE_E2E_ADMIN_PASSWORD }}
+  BRANCH: ${{ github.head_ref || github.ref_name }} 
+  COMMIT_HASH: ${{ github.sha }}
```
Comment (toninis) [inline .github/workflows/e2e-detox-template.yml:65]: No need to add those . It's best to calculate them during the run to ensure we have the proper values per scenario (scheduled, external trigger , fork ..)

File: .github/workflows/e2e-detox-template.yml [hunk 5]
```diff
@@ -107,13 +115,13 @@ jobs:
...
-          device_name: ${{ steps.resolve-device.outputs.DEVICE_NAME }}
-          device_os_version: ${{ steps.resolve-device.outputs.DEVICE_OS_VERSION }}
+          device_name: ${{ env.DEVICE_NAME }}
+          device_os_version: ${{ env.DEVICE_OS_VERSION }}
 
   e2e-ios:
     name: ios-detox-e2e-${{ matrix.runId }}-${{ matrix.deviceName }}-${{ matrix.deviceOsVersion }}
     if: ${{ inputs.run-ios-tests }}
-    runs-on: macos-14
+    runs-on: macos-12
```
Comment (toninis) [inline .github/workflows/e2e-detox-template.yml:124]: Why this change ? macos-14 is faster and M1

File: .github/workflows/e2e-detox-template.yml [hunk 6]
```diff
@@ -205,12 +219,10 @@ jobs:
           DETOX_AWS_SECRET_ACCESS_KEY: ${{ secrets.MM_MOBILE_DETOX_AWS_SECRET_ACCESS_KEY }}
           IOS: ${{ inputs.run-ios-tests }}
           BUILD_ID: ${{ needs.generate-specs.outputs.build_id }}
-          DEVICE_NAME: ${{ needs.generate-specs.outputs.device_name }}
-          DEVICE_OS_VERSION: ${{ needs.generate-specs.outputs.device_os_version }}
           REPORT_PATH: ${{ steps.s3.outputs.path }}
           ## These are needed for the MM Webhook report
           COMMIT_HASH: ${{ needs.generate-specs.outputs.mobile_sha }}
-          BRANCH: ${{ needs.generate-specs.outputs.mobile_ref }}
+          BRANCH: ${{ env.BRANCH }}
```
Comment (toninis) [inline .github/workflows/e2e-detox-template.yml:225]: No need for this change as stated above

**Review by toninis — CHANGES_REQUESTED** [review]

**Comment by toninis** [comment]
Overall great work 🚀  
Some minor improvements for the pipelines

File: .github/workflows/e2e-detox-template.yml (hunk 5 above)
Comment (yasserfaraazkhan) [reply .github/workflows/e2e-detox-template.yml:124]: My Bad, I rebased and kept current changes

File: .github/workflows/e2e-detox-template.yml (hunk 3 above)
Comment (yasserfaraazkhan) [reply .github/workflows/e2e-detox-template.yml:72]: I've added this back

**Review by toninis — APPROVED** [review]
